datafusion/core/tests/sql/decimal.rs

```diff
@@ -0,0 +1,428 @@
+        "| 0.000020 | 0.000000000002 | 3  | false | 0.0000190 |",
+        "| 0.000030 | 0.000000000003 | 5  | true  | 0.0000110 |",
+        "| 0.000050 | 0.000000000005 | 8  | false | 0.0000330 |",
+        "+----------+----------------+----+-------+-----------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+    Ok(())
+}
+
+#[tokio::test]
+async fn decimal_agg_function() -> Result<()> {
+    let ctx = SessionContext::new();
+    register_decimal_csv_table_by_sql(&ctx).await;
+    // min
+    let sql = "select min(c1) from decimal_simple where c4=false";
+    let actual = execute_to_batches(&ctx, sql).await;
+    assert_eq!(
+        &DataType::Decimal(10, 6),
+        actual[0].schema().field(0).data_type()
+    );
+    let expected = vec![
+        "+------------------------+",
+        "| MIN(decimal_simple.c1) |",
+        "+------------------------+",
+        "| 0.000020               |",
+        "+------------------------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+    // max
+    let sql = "select max(c1) from decimal_simple where c4=false";
+    let actual = execute_to_batches(&ctx, sql).await;
+    assert_eq!(
+        &DataType::Decimal(10, 6),
+        actual[0].schema().field(0).data_type()
+    );
+    let expected = vec![
+        "+------------------------+",
+        "| MAX(decimal_simple.c1) |",
+        "+------------------------+",
+        "| 0.000050               |",
+        "+------------------------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+
+    // sum
+    let sql = "select sum(c1) from decimal_simple";
+    let actual = execute_to_batches(&ctx, sql).await;
+    // inferred precision is 10+10
+    assert_eq!(
+        &DataType::Decimal(20, 6),
+        actual[0].schema().field(0).data_type()
+    );
+    let expected = vec![
+        "+------------------------+",
+        "| SUM(decimal_simple.c1) |",
+        "+------------------------+",
+        "| 0.000550               |",
+        "+------------------------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+
+    // avg
+    let sql = "select avg(c1) from decimal_simple";
+    let actual = execute_to_batches(&ctx, sql).await;
+    assert_eq!(
+        &DataType::Decimal(14, 10),
+        actual[0].schema().field(0).data_type()
+    );
+    let expected = vec![
+        "+------------------------+",
+        "| AVG(decimal_simple.c1) |",
+        "+------------------------+",
+        "| 0.0000366666           |",
+        "+------------------------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn decimal_logic_op() -> Result<()> {
+    let ctx = SessionContext::new();
+    register_decimal_csv_table_by_sql(&ctx).await;
+    // logic operation: eq
+    let sql = "select * from decimal_simple where c1=0.00002";
+    let actual = execute_to_batches(&ctx, sql).await;
+    assert_eq!(
+        &DataType::Decimal(10, 6),
```

Comment:
is there a reason this test only checks the schema of `c1` and not `c2`? 

It also isn't clear to me why this test is checkout the output schema but `decimal_by_filter` isn't (not that this is a problem I am just curious)

Reply:
thanks for your point.
I will add the checker for the output schema in all test cases.

Reply:
> is there a reason this test only checks the schema of c1 and not c2?
this test just wants to verify the decimal data type, I will add the test for c5.